addon/components/sidenotes-wrapper.js

```diff
@@ -0,0 +1,89 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { debounce } from '@ember/runloop';
+
+import { getNotesIdealPlacement } from '../utils/notes-placement';
+
+export default class SidenotesWrapperComponent extends Component {
+  @tracked _selectedSidenoteId = null;
+  @tracked _alignedSidenoteId = null;
+
+  @tracked notes = [];
+
+  get gutter() {
+    return this.args.gutter ?? 0;
+  }
+
+  get controlledMode() {
+    return this.args.selectedSidenoteId !== undefined;
+  }
+
+  constructor() {
+    super(...arguments);
+
+    this._selectedSidenoteId = this._alignedSidenoteId = this.controlledMode
+      ? this.args.selectedSidenoteId
+      : this.args.defaultSidenoteId ?? null;
+  }
+
+  placeNotes() {
+    const idealPlacement = getNotesIdealPlacement(
+      this.notes,
+      this._alignedSidenoteId,
+      this.gutter
+    );
+
+    idealPlacement.forEach(({ top, note: { element } }) => {
+      element.style.top = `${top}px`;
+    });
+
+    this.args.onSidenotesMoved?.();
+  }
+
+  @action
+  onRegisterSidenote({ element, id, offsetY, data }) {
```

Comment:
Why use destructuring as you recreate the same object on next line?

Reply:
https://github.com/concordnow/ember-sidenotes/pull/1/commits/e1638dafae73ae8cd3c2330f0d54f66511229d1b :+1: